src/python/pants/engine/isolated_process.py

```diff
@@ -23,28 +24,25 @@ class ExecuteProcessRequest(datatype([
   ('argv', tuple),
   ('input_files', DirectoryDigest),
   ('description', text_type),
-  ('env', tuple),
-  ('output_files', tuple),
-  ('output_directories', tuple),
+  # TODO: allow inferring a default value if a type is provided which has a 0-arg constructor.
```

Comment:
I would be happy to just remove this TODO; inferring defaults is scary...

Reply:
After discussion on slack, agreed and will do.